samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -167,15 +167,18 @@ object JobModelManager extends Logging {
     */
   def getProcessorLocality(config: Config, localityManager: LocalityManager) = {
```

Comment:
Why do we need this here can't the callee of this not call LocalityManager since we are already making them independent

Reply:
discussed offline with Bharath since JobModelMAnager dumping locality info when LocalityManager is kind of independent does not make sense to me, please cut a jira here and add a todo so we can track this cleanup

Reply:
Putting the points from our offline discussion. This is a helper method within the `JobModelManager` that does some massaging on the locality information and decorates it with its own terminology for the absence of locality `ANY_HOST`. The helper method could carry a perception that this is just getting locality information but in theory it is not. Also, this goes over the list of all available containers and populates `ANY_HOST` for the ones that don't have locality. This information is oblivious to the `LocalityManager` as it only treats coordinator stream as the source of truth which only knows of container that has locality data.